internal/bodyprocessors/xml.go

```diff
@@ -15,17 +16,64 @@ type xmlBodyProcessor struct {
 }
 
 func (*xmlBodyProcessor) ProcessRequest(reader io.Reader, v plugintypes.TransactionVariables, options plugintypes.BodyProcessorOptions) error {
-	values, contents, err := readXML(reader)
+	// Read the entire body to store it raw
+	buf := new(bytes.Buffer)
+	tee := io.TeeReader(reader, buf)
+
+	// Process XML as before
+	values, contents, err := readXML(tee)
 	if err != nil {
 		return err
 	}
+
+	// Store standard structure
 	col := v.RequestXML()
 	col.Set("//@*", values)
 	col.Set("/*", contents)
+
+	// Store the raw XML in the TX variable for validateSchema to use
+	rawXml := buf.String()
+	if txVar := v.TX(); txVar != nil && v.RequestBody() != nil {
+		// Store the content type and raw body
+		txVar.Set("xml_request_body", []string{rawXml})
```

Comment:
is this how modsec handles this?
Otherwise, I think we should use the transaction context, as it becomes a future concern for data masking

Reply:
Reverted this change.